pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -0,0 +1,213 @@
+
+    def check_errors(self):
+        return self.instrument.check_errors()
+
+    def assert_trigger(self):
+        """This method asserts a trigger pulse on one of the digital I/O lines."""
+
+        log.info("Asserting a trigger pulse on pin number %d." % self.pin_number)
+        self.write("assert()")
+        self.check_errors()
+
+    def clear_trigger(self):
+        """This method clears the trigger event detector on a digital I/O line."""
+
+        log.info("Clearing trigger on pin number %d." % self.pin_number)
+        self.write("clear()")
+        self.check_errors()
+
+    def get_event_id(self):
+        """This method returns the mode in which the trigger event detector and
+        the output trigger generator operate on the given trigger line. See description
+        of all the possible EVENT_IDs on page 9-57 of the Series 2600B Reference Manual.
+        """
+
+        id = self.ask("EVENT_ID")
+        self.check_errors()
+        return int(id)
+
+    def get_overrun_status(self):
+        """This method returns the event detector overrun status. If this is
+        true, an event was ignored because the event detector was already in the
+        detected state when the event occurred. This is an indication of the
+        state of the event detector built into the line itself. It does not
+        indicate if an overrun occurred in any other part of the trigger model
+        or in any other detector that is monitoring the event."""
+
+        response_status = self.ask("overrun")
+        self.check_errors()
+
+        if response_status == "false":
+            status = False
+        elif response_status == "true":
+            status = True
+        else:
+            status = None
+
+        return status
+
+    def release_trigger(self):
+        """This method releases an indefinite length or latched trigger."""
+
+        log.info("Releasing trigger on pin number %d." % self.pin_number)
+        self.write("release()")
+        self.check_errors()
+
+    def reset_trigger_values(self):
+        """This method resets trigger values to their factory defaults. It
+        sets the mode, pulsewidth, stimulus, and overrun status to factory
+        default settings.
+        """
+
+        log.info(
+            "Resetting trigger values (to factory defaults) on pin number %d."
+            % self.pin_number
+        )
+        self.write("reset()")
+        self.check_errors()
+
+    def wait_for_trigger(self, timeout):
+        """This method waits for a trigger for up to a maximum of the timeout value (in seconds).
+        Returns True if a trigger was detected, false if the timout was reached and no trigger was
+        detected.
+        """
+
+        log.info(
+            "Waiting for trigger for %f on pin number %d." % (timeout, self.pin_number)
+        )
+        self.write("wait(timeout)")
+        self.check_errors()
+
+    trigger_mode = Instrument.control(
```

Comment:
Is there a way to return this property as an integer? I can't cast the return value in my `ask` method as it might needs to return floats, integers, and strings, depending on the wrapper method used.

Reply:
With "map_values" you will turn strings to integers on setting and integers to strings on getting. Right now, you cannot separate those two directions directly.

As a hack, you could have a validator, which looks up the string in the dictionary and returns the integer. Now you can set map_values to False (default) and you will get integers.

At least that is how I understand `Instrument.control`.

Reply:
I didn't realize mapping worked both with getting and setting - in this case, I want the user to get the string description (in this case, the dictionary keys) when getting this property, so I can leave it the way it is. Thank you @bmoneke !